core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -8230,24 +8231,55 @@ private void checkLiteral2(String expression, String expected) {
   }
 
   /** Test case for
-   * <a href="https://issues.apache.org/jira/browse/CALCITE-6819">[CALCITE-6819]
-   * MSSQL doesn't support TRUE/FALSE keywords in its Join predicate</a>. */
-  @Test void testJoinBoolLiteralMSSQL() {
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6847">[CALCITE-6847]
+   * Support ClickHouse Dialect with join on Boolean type</a>. */
+  @Test void testJoinBoolLiteral() {
     final String queryTrue = "SELECT \"hire_date\", \"department_description\" FROM \"employee\" "
         + "LEFT JOIN \"department\" ON TRUE";
+    final String queryFalse = "SELECT \"hire_date\", \"department_description\" FROM \"employee\" "
+        + "LEFT JOIN \"department\" ON False";
+
+    final String queryTrue1 = "SELECT \"hire_date\", \"department_description\" FROM \"employee\" "
+        + "LEFT JOIN \"department\" ON 1 = 1";
+    final String queryFalse1 = "SELECT \"hire_date\", \"department_description\" FROM \"employee\" "
+        + "LEFT JOIN \"department\" ON 1 = 0";
+
+    // mssql test
     final String mssqlExpected1 = "SELECT [employee].[hire_date],"
         + " [department].[department_description]\nFROM [foodmart].[employee]\nLEFT JOIN"
         + " [foodmart].[department] ON (1 = 1)";
     sql(queryTrue)
         .dialect(MssqlSqlDialect.DEFAULT).ok(mssqlExpected1);
+    sql(queryTrue1)
+        .dialect(MssqlSqlDialect.DEFAULT).ok(mssqlExpected1);
+
 
-    final String queryFalse = "SELECT \"hire_date\", \"department_description\" FROM \"employee\" "
-        + "LEFT JOIN \"department\" ON False";
     final String mssqlExpected2 = "SELECT [employee].[hire_date],"
         + " [department].[department_description]\nFROM [foodmart].[employee]\nLEFT JOIN"
         + " [foodmart].[department] ON (1 = 0)";
     sql(queryFalse)
         .dialect(MssqlSqlDialect.DEFAULT).ok(mssqlExpected2);
+    sql(queryFalse1)
+        .dialect(MssqlSqlDialect.DEFAULT).ok(mssqlExpected2);
+
+    // clickhouse test
+    final String clickhouseExpected1 = "SELECT `employee`.`hire_date`,"
```

Comment:
How about this SQL:
```
SELECT TRUE from "employee"
```

Reply:
@NobiGo would error out as:
**select true from system.numbers limit 1;**
Received exception from server (version 21.8.11):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: 'true' while processing query: 'SELECT true FROM system.numbers LIMIT 1', required columns: 'true'. 

but can support select 1=1 from system.numbers limit 1;
**select 1=1 from system.numbers limit 1;**
1
1 rows retrieved starting from 1 in 89 ms (execution: 76 ms, fetching: 13 ms)